extensions/objects/models/objects.py

```diff
@@ -1633,208 +1552,265 @@ def get_schema(cls):
         }
 
 
-class TranslatableUnicode(BaseTranslatableObject):
+class TranslatableHtmlContentId(BaseObject):
+    """A TranslatableHtml content id."""
+
+    default_value = ''
+
+    @classmethod
+    def get_schema(cls):
+        """Returns the object schema.
+
+        Returns:
+            dict. The object schema.
+        """
+        return UnicodeString.get_schema()
+
+
+class SetOfTranslatableHtmlContentIds(BaseObject):
+    """A Set of TranslatableHtml content ids."""
+
+    default_value = []
+
+    @classmethod
+    def get_schema(cls):
+        """Returns the object schema.
+
+        Returns:
+            dict. The object schema.
+        """
+        return {
+            'type': 'list',
+            'items': TranslatableHtmlContentId.get_schema(),
+            'validators': [{
+                'id': 'is_uniquified'
+            }]
+        }
+
+
+class ListOfSetsOfTranslatableHtmlContentIds(BaseObject):
+    """List of sets of TranslatableHtml content ids."""
+
+    default_value = []
+
+    @classmethod
+    def get_schema(cls):
+        """Returns the object schema.
+
+        Returns:
+            dict. The object schema.
+        """
+        return {
+            'type': 'list',
+            'items': SetOfTranslatableHtmlContentIds.get_schema()
+        }
+
+
+class BaseTranslatableObject(BaseObject):
+    """Base translatable object class.
+
+    This is a superclass for objects that are translatable and thus require a
+    content id. This class enforces that the object is a dictionary with a
+    content id field. The schema of the actual value is determined by the
+    _value_schema property.
+    """
+
+    # The key name in the translatable object corresponding to the translatable
+    # value. This field must be populated by subclasses.
+    _value_key_name = None
+    # The schema of the translatable value. This field must be populated by
+    # subclasses.
+    _value_schema = None
+    # The default value of the object. This field must be populated by
+    # subclasses.
+    default_value = None
+
+    @staticmethod
+    def normalize_value(value):
```

Comment:
Why is this made public, any specific reason? (I think we should only have the `normalize` function as public)

Reply:
Aha, this was a big realization I had in this PR. When validating the stuff in the written_translations, I only actually want to validate the translation value and not the content IDs. So this function needs to be public so that core/domain can call it directly.

The original proposal was to get rid of SubtitledHtml and SubtitledUnicode completely and turn them into these translatable objects instead (see #11433) but on reflection I think that that would be too large of a revamp and that those domain objects are useful. So instead I'm working on the seam between data-format and translatable-objects instead.